app/src/main/java/br/com/mgalhardo/guidebook/dagger/UiComponent.java

```diff
@@ -0,0 +1,12 @@
+package br.com.mgalhardo.guidebook.dagger;
+
+import br.com.mgalhardo.guidebook.dagger.module.presentation.PresenterModule;
+import br.com.mgalhardo.guidebook.dagger.scope.PerActivity;
+import br.com.mgalhardo.guidebook.presentation.ui.guidelist.GuideListActivity;
+import dagger.Subcomponent;
+
+@PerActivity
```

Comment:
Yes. Since the presenter hold a strong reference of the view (MVP), It's better to use a component with a scope that is destroyed with the activity to avoid leakages. This rule apply to any dependency that rely on activities. Also the presenter will not be used anywhere else.